scripts/visualize_learning_rate.py

```diff
@@ -105,7 +120,43 @@ def visualize(learning_rates):
             "epochs value in the cofig file in 'config.trainer.epochs' or using '--epochs' argument.")
         epochs = 100
 
-    lrs = calculate_lr(lr_scheduler_cfg, epochs=epochs, optimizer_config=optim_cfg)
+    # Get accumulation step
+    accumulation_step = 1
+    
+    ## Try to get accumulation_step from experiment config
+    try:
+        accumulation_step = config.trainer.driver.args.accumulation_step
+    except:
+        pass
+
+    ## Override if accumulation step described in args
+    if args.accumulation_step is not None:
+        accumulation_step = args.accumulation_step
+
+    # Get steps per epoch
+
+    steps_per_epoch = 100
+
+    ## Try to get accumulation_step from experiment config
+    try:
+        dataloader = create_dataloader(dataloader_config=config.dataloader,
+                                        dataset_config=config.dataset,
+                                        preprocess_config = config.model.preprocess_args,
+                                        collate_fn=None)
+        steps_per_epoch = len(dataloader)
+    except:
+        pass
+    ## Override if steps_per_epoch described in args
+    if args.steps_per_epoch is not None:
```

Comment:
why not take the args first rather than create a dataloader which is possibly be unused?

Reply:
ah good points